### ISSUE TITLE
training set, validation set, and  test set

### ISSUE THREAD

**Issue by chenyongpeng1** [biomodhub/biomod2#475]
Hello， biomod2 team！
 Species records are randomly divided into a training set (75% of the data) for model calibration and a test set (25% of the data) for validation. 

--This is somewhat misleading. Typically there should be three dataset parts: : a training set, a validation set, and a test set. The validation set is used to fine-tune the model's hyperparameters, while the test set is used to assess the model accuracy to unseen data.  
Whether biomod2 contains validation sets. If not, can you add a validation set and use techniques such as cross-validation to mitigate the overfitting problem?

**Comment by MayaGueguen** [comment]
Hello Chenyong :wave: 

Actually, we do have _calibration_, _validation_ and _evaluation_ datasets :slightly_smiling_face: 
- while _calibration_ and _validation_ will come from your original dataset
- _evaluation_ is a completely different dataset that will only be used, if provided, to compute evaluation metrics using models built over the original dataset

Datasets are given to the [BIOMOD_FormatingData](https://biomodhub.github.io/biomod2/reference/BIOMOD_FormatingData.html) : 
- original dataset is given through `resp.var`, `expl.var`, `resp.xy` parameters
- while _evaluation_ dataset is provided through `eval.resp.var`, `eval.expl.var`, `eval.resp.xy` parameters

Cut of the original dataset into _calibration_ and _validation_ is done within [BIOMOD_Modeling](https://biomodhub.github.io/biomod2/reference/BIOMOD_Modeling.html) function through `CV.[...]` parameters. (_This is done by calling the [bm_CrossValidation](https://biomodhub.github.io/biomod2/reference/bm_CrossValidation.html) function, which you can also use by your own if necessary_)

:information_source: Note that there are several possibilities to build your calibration and validation datasets. You can find more details within the [Cross-validation vignette](https://biomodhub.github.io/biomod2/articles/vignette_crossValidation.html).

When calling to [get_evaluations](https://biomodhub.github.io/biomod2/reference/getters.out.html) function to retrieve your evaluation values, you will see 3 columns in your output table : _calibration_, _validation_ and _evaluation_. They will only be filled if the corresponding dataset was provided, so _evaluation_ will be empty if you did not provide evaluation dataset to [BIOMOD_FormatingData](https://biomodhub.github.io/biomod2/reference/BIOMOD_FormatingData.html), and _validation_ column will be empty if for example you ask to build a model with all the data (`CV.do.full.models = TRUE` in [BIOMOD_Modeling](https://biomodhub.github.io/biomod2/reference/BIOMOD_Modeling.html)).



:arrow_right: Note also that we have plenty of tutorial and documentation onto our [website](https://biomodhub.github.io/biomod2/index.html), and you can have an overview of the package functions within this [presentation](https://biomodhub.github.io/biomod2/articles/vignette_presentation.html). So do not hesitate to have a look :eyes: 

Hope it helps,
Maya

**Comment by chenyongpeng1** [comment]
Hello，Maya
Thank you for your patience and attention, and I have studied a lot.
but I still have some questions：
1. Can the original data set be automatically divided into three parts through the CV Function, for example, one part is calibration, one part is validation, and one part is test or evaluation according to the ratio of 70:10:20？and  `eval.resp.var, eval.expl.var, eval.resp.xy` parameters  are clearly the same as before.
2. For the TSS value, is it computed from the all data set, or from the calibration or validation data set?TSS=sensitivity+specificity-1 ,but I found that calibration value=sensitivity+specificity-1 for TSS, Is this right?
Looking forward to your reply! Thanks again!

Best wishes,
 Chenyongpeng

**Comment by MayaGueguen** [comment]
Hello Chenyongpeng,

> Can the original data set be automatically divided into three parts through the CV Function, for example, one part is calibration, one part is validation, and one part is test or evaluation according to the ratio of 70:10:20？and eval.resp.var, eval.expl.var, eval.resp.xy parameters are clearly the same as before.

No. As the `evaluation` dataset is supposed to be independant data from the one used to build the models, original data can only be divided in `calibration` and `validation` through biomod2.

> I'm a little confused：biomod2 allows you to use different strategies to separate your data into a calibration dataset and a validation dataset for the cross-validation. But I have confused that，in this picture，which is validation set？In other words，validation dataset in Biomod2 is equivalent to validation fold from training set，or test set in this picture.

In your picture : 
- Training Data set = what I called original data in previous answer
     -  Training fold = `calibration` dataset in biomod2
     - Validation fold = `validation` dataset in biomod2
- Test Data set = `evaluation` dataset in biomod2

Hope it helps,
Maya

**Comment by MayaGueguen** [comment]
@chenyongpeng1 

If you want to have an illustration of biomod2 difference between _calibration_, _validation_ and _evaluation_ datasets, we just published a [new video](https://biomodhub.github.io/biomod2/articles/vignette_videos.html) :movie_camera: on our website. You might want to check the `02. Datasets` part, at 4:14 :eyes:
